lib/gather-fragment-imports.js

```diff
@@ -0,0 +1,91 @@
+
+/**
+ * Helper function to fetch and parse all the direct and transitive imports starting
+ * from the source. Return object is:
+ *
+ * {
+ *     1: {
+ *         FooBar: fragment,
+ *         Bar: fragment
+ *     }
+ * }
+ *
+ * @param {string} source - .graphql source code
+ * @param {string} sourceLocation - File path of the source code
+ * @param {function} resolveImport - This function is used to resolve the path to file for import
+ * @param {function} fragmentParserGenerator - Generator function that is passed source code. Expected to yield objects for each fragment definition
+ * @param {boolean} throwIfImportNotFound - If set to true, throws error if import not found
+ */
+function gatherFragmentImports({
+  source,
+  sourceLocation,
+  resolveImport,
+  fragmentParserGenerator,
+  throwIfImportNotFound,
+}) {
+  /**
+   * {
+   *     1: {
+   *         FooBar: fragment,
+   *         Bar: fragment
+   *     }
+   * }
+   * @type {Map<number, Map<string, object>>}
+   */
+  const lineToFragmentDefinitions = new Map();
+  const importLinesToInlinedSource = inlineImports.lineToImports(source, {
+    resolveOptions: {
+      basedir: path.dirname(sourceLocation),
+    },
+    resolveImport,
+    throwIfImportNotFound,
+  });
+
+  for (const [lineNumber, source] of importLinesToInlinedSource) {
+    const fragmentDefinitionsBucket = lineToFragmentDefinitions[lineNumber] || new Map();
+    for (const fragment of fragmentParserGenerator(source)) {
+      fragmentDefinitionsBucket.set(fragment.name.value, fragment);
+    }
+    lineToFragmentDefinitions.set(lineNumber, fragmentDefinitionsBucket);
+  }
+
+  return lineToFragmentDefinitions;
+}
+
+/**
+ * Wrapper function to setup most parameters from the context object to call gatherFragmentImports.
+ *
+ * @param {object} context - Context object that was passed to the `create` function of an eslint rule
+ * @param {boolean} throwIfImportNotFound - If set to true, throws error if import not found
+ * @returns {Map<number, Map<string, Object>>}
+ */
+function gatherFragmentImportsForContext(context, throwIfImportNotFound) {
+  let resolveImport;
+
+  let importResolver =
+    Array.isArray(context.options) &&
+    context.options.length > 0 &&
+    context.options[0].importResolver;
+  if (typeof importResolver === 'string') {
+    if (!path.isAbsolute(importResolver)) {
+      throw new Error(`Option "importResolver" must be an absolute path, not "${importResolver}"`);
+    }
+
+    resolveImport = require(importResolver);
+  } else {
+    resolveImport = require('resolve').sync;
+  }
+
+  return gatherFragmentImports({
+    source: context.getSourceCode().text,
+    sourceLocation: context.getFilename(),
+    resolveImport,
+    fragmentParserGenerator: context.parserServices.getFragmentDefinitionsFromSource,
+    throwIfImportNotFound,
+  });
+}
+
+module.exports = { gatherFragmentImports, gatherFragmentImportsForContext };
```

Comment:
re https://github.com/stefanpenner/graphql-fragment-import/pull/9#discussion_r620497270

this looks great, but what do you think about having this function specifically be in the eslint package and not the general lib package, since it's a helper for eslint rules specifically and not really relevant for e.g. command-line usage?